src/main/java/com/iemr/hwc/service/ncdCare/NCDCareServiceImpl.java

```diff
@@ -266,47 +267,124 @@ public Map<String, Long> saveBenVisitDetails(JsonObject visitDetailsOBJ, CommonU
-			Long benVisitCode = commonNurseServiceImpl.generateVisitCode(benVisitID, nurseUtilityClass.getVanID(),
-					nurseUtilityClass.getSessionID());
-
-			if (benVisitID != null && benVisitID > 0 && benVisitCode != null && benVisitCode > 0) {
-				if (visitDetailsOBJ.has("adherence") && !visitDetailsOBJ.get("adherence").isJsonNull()) {
-					// Save Ben Adherence
-					BenAdherence benAdherence = InputMapper.gson().fromJson(visitDetailsOBJ.get("adherence"),
-							BenAdherence.class);
-					benAdherence.setBenVisitID(benVisitID);
-					benAdherence.setVisitCode(benVisitCode);
-					adherenceSuccessFlag = commonNurseServiceImpl.saveBenAdherenceDetails(benAdherence);
-				}
-				if (visitDetailsOBJ.has("investigation") && !visitDetailsOBJ.get("investigation").isJsonNull()) {
-					// Save Ben Investigations
-					WrapperBenInvestigationANC wrapperBenInvestigationANC = InputMapper.gson()
-							.fromJson(visitDetailsOBJ.get("investigation"), WrapperBenInvestigationANC.class);
-
-					if (wrapperBenInvestigationANC != null) {
-						wrapperBenInvestigationANC.setBenVisitID(benVisitID);
-						wrapperBenInvestigationANC.setVisitCode(benVisitCode);
-						investigationSuccessFlag = commonNurseServiceImpl
-								.saveBenInvestigationDetails(wrapperBenInvestigationANC);
-
-					} else {
-						// Invalid Data..
+			int i = commonNurseServiceImpl.getMaxCurrentdate(benVisitDetailsOBJ.getBeneficiaryRegID(),
+					benVisitDetailsOBJ.getVisitReason(), benVisitDetailsOBJ.getVisitCategory());
+			if (i < 1) {
+				benVisitID = commonNurseServiceImpl.saveBeneficiaryVisitDetails(benVisitDetailsOBJ);
+
+				// 11-06-2018 visit code
+				Long benVisitCode = commonNurseServiceImpl.generateVisitCode(benVisitID, nurseUtilityClass.getVanID(),
+						nurseUtilityClass.getSessionID());
+
+				if (benVisitID != null && benVisitID > 0 && benVisitCode != null && benVisitCode > 0) {
+					if (visitDetailsOBJ.has("adherence") && !visitDetailsOBJ.get("adherence").isJsonNull()) {
+						// Save Ben Adherence
+						BenAdherence benAdherence = InputMapper.gson().fromJson(visitDetailsOBJ.get("adherence"),
+								BenAdherence.class);
+						benAdherence.setBenVisitID(benVisitID);
+						benAdherence.setVisitCode(benVisitCode);
+						adherenceSuccessFlag = commonNurseServiceImpl.saveBenAdherenceDetails(benAdherence);
+					}
+					if (visitDetailsOBJ.has("investigation") && !visitDetailsOBJ.get("investigation").isJsonNull()) {
+						// Save Ben Investigations
+						WrapperBenInvestigationANC wrapperBenInvestigationANC = InputMapper.gson()
+								.fromJson(visitDetailsOBJ.get("investigation"), WrapperBenInvestigationANC.class);
+
+						if (wrapperBenInvestigationANC != null) {
+							wrapperBenInvestigationANC.setBenVisitID(benVisitID);
+							wrapperBenInvestigationANC.setVisitCode(benVisitCode);
+							investigationSuccessFlag = commonNurseServiceImpl
+									.saveBenInvestigationDetails(wrapperBenInvestigationANC);
+
+						} else {
+							// Invalid Data..
```

Comment:
The comment sounds like it is a grave problem, but there is no code that handles it.